lib/analysis.js

```diff
@@ -80,6 +82,16 @@ AnalysisFactory.prototype.create = function(configuration, definition, callback)
                     return done(err, analysis);
                 });
             },
+            function analysis$collectRequirements(analysis, done) {
+                requirements.computeRequirements(analysis, function(err) {
+                    return done(err, analysis);
+                });
+            },
+            function analysis$validateRequirements(analysis, done) {
```

Comment:
I'd check limits after validating analysis and before registering it. If some node reaches the limit the analysis should fail and it shouldn't be registered.